platform/security/idp/security-idp-server/src/main/java/org/codice/ddf/security/idp/server/IdpEndpoint.java

```diff
@@ -0,0 +1,820 @@
+                    token = handlerResult.getToken();
+                }
+            } catch (ServletException e) {
+                LOGGER.warn("Encountered an exception while checking for PKI auth info.", e);
+            }
+        } else if (USER_PASS.equals(authMethod)) {
+            BasicAuthenticationHandler basicAuthenticationHandler = new BasicAuthenticationHandler();
+            HandlerResult handlerResult = basicAuthenticationHandler
+                    .getNormalizedToken(request, null, null, false);
+            if (handlerResult.getStatus().equals(HandlerResult.Status.COMPLETED)) {
+                token = handlerResult.getToken();
+            }
+        } else if (GUEST.equals(authMethod)) {
+            token = new AnonymousAuthenticationToken(BaseAuthenticationToken.ALL_REALM);
+        } else {
+            throw new IllegalArgumentException("Auth method is not supported.");
+        }
+
+        org.w3c.dom.Element samlToken = null;
+        String statusCode;
+        try {
+            Subject subject = securityManager.getSubject(token);
+            for (Object principal : subject.getPrincipals().asList()) {
+                if (principal instanceof SecurityAssertion) {
+                    SecurityToken securityToken = ((SecurityAssertion) principal)
+                            .getSecurityToken();
+                    samlToken = securityToken.getToken();
+                }
+            }
+            statusCode = StatusCode.SUCCESS_URI;
+        } catch (SecurityServiceException e) {
+            if (!passive) {
+                throw e;
+            } else {
+                statusCode = StatusCode.AUTHN_FAILED_URI;
+            }
+        }
+
+        return SamlProtocol.createResponse(
+                SamlProtocol.createIssuer("https://localhost:8993/services/idp/login"),
+                SamlProtocol.createStatus(statusCode), authnRequest.getID(), samlToken);
+    }
+
+    protected void signSamlPostResponse(org.opensaml.saml2.core.Response samlResponse)
+            throws WSSecurityException, SimpleSign.SignatureException {
+        SimpleSign.signSamlObject(samlResponse, getSignatureCrypto(),
+                getSignatureCrypto().getDefaultX509Identifier(), getSignaturePassword());
+    }
+
+    protected URI signSamlGetResponse(org.opensaml.saml2.core.Response samlResponse,
+            AuthnRequest authnRequest, String relayState)
+            throws WSSecurityException, SimpleSign.SignatureException, IOException {
+        Document doc = DOMUtils.createDocument();
+        doc.appendChild(doc.createElement("root"));
+        String encodedResponse = URLEncoder.encode(RestSecurity.deflateAndBase64Encode(
+                DOM2Writer.nodeToString(OpenSAMLUtil.toDom(samlResponse, doc, false))), "UTF-8");
+        String requestToSign =
+                SSOConstants.SAML_RESPONSE + "=" + encodedResponse + "&" + SSOConstants.RELAY_STATE
```

Comment:
Same comment re: `String.format`. If you choose to ignore my previous one, then this one should be ignored as well. :)